optee: fix tee-supplicant abort when receive KILLSIG

airbak opened the pull request:
When press ctrl-c to CA, it will receive KILLSIG singal.
And the wait_for_completion_interruptible will always been true.
But actully not tee-supplicant receive KILLSIG, so it become an
endless loop.
To solve this issue:
1)Check whether tee-supplicant is available before create a request.
2)Replace wait_for_completion_interruptible with wait_for_completion.

Signed-off-by: Jianhui Li <airbak.li@hisilicon.com>

jenswikl commented on drivers/tee/optee/supp.c:
```diff
@@ -88,10 +88,15 @@ u32 optee_supp_thrd_req(struct tee_context *ctx, u32 func, size_t num_params,
 
+	if (!supp->ctx) {
+		pr_err("There's no tee-supplicant available, please start it\n");
+		return TEEC_ERROR_COMMUNICATION;
```
This doesn't deal with the case during boot where tee-supplicant may not have started yet so clients may get unexpected errors with no clear way of dealing with them.

airbak replied:
If tee-supplicant have not started yet,  it also show this error message.
And ca open session failed with error code 0xffff000e(TEE_ERROR_COMMUNICATION).

jenswikl replied:
Yes, but what's the client supposed to do about that? Try again and again and hope that an unspecific error goes away?

airbak replied:
You are right, close it.

Commit:
optee: fix tee-supplicant abort when receive KILLSIG

When press ctrl-c to CA, it will receive KILLSIG singal.
And the wait_for_completion_interruptible will always been true.
But actully not tee-supplicant receive KILLSIG, so it become an
endless loop.
To solve this issue:
1)Check whether tee-supplicant is available before create a request.
2)Replace wait_for_completion_interruptible with wait_for_completion.

Signed-off-by: Jianhui Li <airbak.li@hisilicon.com>
